app/Console/Commands/AddCentre.php

```diff
@@ -0,0 +1,128 @@
+                                {email : email address of User who\'s responsible for this action}
+                                ';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Adds a Children\'s Centre to a sponsoring entity';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        // Find user
+        $this->user = User::where('email', $this->argument('email'))
+            ->first();
+
+        // Find the pre-exisiting sponsor.
+        $this->sponsor = Sponsor::where('shortcode', $this->argument('shortcode'))
+            ->first();
+
+        // Check if there's a pre-existing centre.
+        $this->centre = Centre::where('prefix', $this->argument('prefix'))
+            ->orWhere('name', $this->argument('name'))
+            ->first();
+
+        $this->preferences = Centre::select('print_pref')->distinct()->pluck('print_pref')->toArray();
+
+        switch (true) {
+            case (!$this->user):
+                exit("Can't find that User.\n");
+                break;
+            case (!$this->sponsor):
+                exit("Sponsor " .
+                    $this->argument('shortcode') .
+                    " does not exist, exit without change.\n");
+            case ($this->centre):
+                exit("Centre " .
+                    $this->centre->name .
+                    ", " .
+                    $this->centre->prefix .
+                    " exists, exit without change.\n");
+            case (!in_array($this->argument('preference'), $this->preferences)):
+                exit("The preference " .
+                    $this->argument('preference') .
+                    " does not exist, exit without change.\n");
+            default:
+                // Check the user is happy to proceed
+                if (!$this->warnUser()) {
+                    exit("Exit without change.\n");
+                };
+                // Log the user in.
+                Auth::login($this->user);
+
+                // Did that work?
+                if (!Auth::check()) {
+                    exit("Failed to login.\n");
+                };
+
+                $this->centre = new Centre([
+                    'name' => $this->argument('name'),
+                    'prefix' => $this->argument('prefix'),
+                    'print_pref' => $this->argument('preference'),
+                ]);
+
+                $this->centre->sponsor()->associate($this->sponsor);
+                $this->centre->save();
+
+                exit("Done.\n");
+        }
+    }
+
+    public function warnUser()
+    {
+        $this->info('This unsafe command will alter the database to add records.');
```

Comment:
That sounds like we want them to say no...

How about: This action will add Centre with name, using code prefix xx to sponsor xx, with pref xx authorised by email.

Reply:
It's a reminder to get your stuff together before you press y. perhaps redundant.

Reply:
Good idea to check - but I think more explicit would be useful rather than just a negative and blind - this is a dumb idea are you sure you want to go through with it.